### PR TITLE
Message Processor Instructions

### PR DESCRIPTION
**Pull request by Ibaqu**
## Purpose
To create instructions to make Message Processor Admin Services

### CODE REVIEW

File: gh-docs/message-processor/Admin_Services.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# How to create admin-services for the Message Processor
```
Comment (a5anka) [inline gh-docs/message-processor/Admin_Services.md]: Shall we make the title more generic? Maybe something like "How to create admin-services for EI"

File: gh-docs/message-processor/Admin_Services.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+### Creation Flow
+
+1. Define your Admin Service function
+2. Generate the stub using wsdl
```
Comment (a5anka) [inline gh-docs/message-processor/Admin_Services.md]: wsdl -> WSDL

File: gh-docs/message-processor/Admin_Services.md (hunk 1 above)
Comment (a5anka) [reply gh-docs/message-processor/Admin_Services.md]: Keeping instructions specific to message processor is OK IMO.

File: gh-docs/message-processor/Admin_Services.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+To generate the stub file, the WSDL file in `<repo-home>/service-stubs/mediation-admin/org.wso2.carbon.message.processor.stub/`
+needs to be edited and then built to generate the stub. 
+
+#### To obtain the contents of the WSDL
+
+ 1. Build 'MessageProcessorAdminService.java' using 'mvn clean install' on the IntelliJ console. 
+ 
+ 2. Copy the generated '.jar' file found in `<repo-home>/components/mediation-admin/org.wso2.carbon.message.processor/target/org.wso2.carbon.message.processor-4.6.106-SNAPSHOT.jar`
+ 
+ 3. Paste the '.jar' file into the 'patches' folder in the product-ui
```
Comment (a5anka) [inline gh-docs/message-processor/Admin_Services.md]: don't we have to create a subdirectory inside patches directory?

File: gh-docs/message-processor/Admin_Services.md (hunk 1 above)
Comment (Ibaqu) [reply gh-docs/message-processor/Admin_Services.md]: We could keep it generic, but I might be missing out on some small details that may not apply in all cases.

File: gh-docs/message-processor/Admin_Services.md (hunk 3 above)
Comment (Ibaqu) [reply gh-docs/message-processor/Admin_Services.md]: Yes. I will mention that

File: gh-docs/Admin_Services.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# How to create admin-services
+Admin Services generally follow the following steps. Here, the Message Processor has been considered as an example
```
Comment (samgnaniah) [inline gh-docs/Admin_Services.md]:
```suggestion
When creating Admin Services you would generally follow the same set of steps. In this instance, the Message Processor is used as an example.
```

File: gh-docs/Admin_Services.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+### Creation Flow
+
+1. Define your Admin Service function
+2. Generate the stub using WSDL
+3. Define function in MessageProcessorAdminServiceClient to call Admin Service function
+4. Call the function from the Message Processor UI jsp
+
+### 1. Defining your Admin Service
+Message Processor Admin Services are defined in the 'MessageProcessorAdminService.java' class in the carbon-mediation repo.
```
Comment (samgnaniah) [inline gh-docs/Admin_Services.md]: Shall we add the link to the carbon-mediation repo?

File: gh-docs/Admin_Services.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
+# How to create admin-services
+Admin Services generally follow the following steps. Here, the Message Processor has been considered as an example
+
+### Creation Flow
+
+1. Define your Admin Service function
+2. Generate the stub using WSDL
+3. Define function in MessageProcessorAdminServiceClient to call Admin Service function
+4. Call the function from the Message Processor UI jsp
```
Comment (samgnaniah) [inline gh-docs/Admin_Services.md]:
```suggestion
4. Call the function from the Message Processor UI JSP
```

File: gh-docs/Admin_Services.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+### 1. Defining your Admin Service
+Message Processor Admin Services are defined in the 'MessageProcessorAdminService.java' class in the carbon-mediation repo.
+
+Go to `<repo-home>/components/mediation-admin/org.wso2.carbon.message.processor/` and navigate to the 'service' directory
+to find the 'MessageProcessorAdminService.java' class.
+
+Once you define your function, you will have to generate the MessageProcessorAdminServiceStub file.
```
Comment (samgnaniah) [inline gh-docs/Admin_Services.md]:
```suggestion
Once you define your function, generate the `MessageProcessorAdminServiceStub` file. 
```

File: gh-docs/Admin_Services.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
+# How to create admin-services
+Admin Services generally follow the following steps. Here, the Message Processor has been considered as an example
+
+### Creation Flow
+
+1. Define your Admin Service function
+2. Generate the stub using WSDL
+3. Define function in MessageProcessorAdminServiceClient to call Admin Service function
```
Comment (samgnaniah) [inline gh-docs/Admin_Services.md]:
```suggestion
3. Define the function in `MessageProcessorAdminServiceClient` to call the Admin Service function
```

Shall we follow a convention of having all class names, file names, method names, etc. between single backticks? This is to highlight this to readers.

File: gh-docs/Admin_Services.md (hunk 5 above)
Comment (Ibaqu) [reply gh-docs/Admin_Services.md]: I am making the necessary changes in a different pull request. I will link the PR here as well once I am done.

File: gh-docs/Admin_Services.md (hunk 8 above)
Comment (Ibaqu) [reply gh-docs/Admin_Services.md]: I will make that change.

**Pull request closed without merging**